haigha/classes/basic_class.py

```diff
@@ -53,8 +54,30 @@ def _cleanup(self):
         self._get_cb = None
         self._recover_cb = None
         self._cancel_cb = None
+        self._return_listener = None
```

Comment:
@awestendorf, that would be easy to change, but I would first like to explain why I chose the different naming for this instance member:
# The other `_*_cb` members are all `deque()`, but `_return_listener` is a single callback, so I was concerned about adding confusion to the code, making things more error-prone to maintain as the result.
# I borrowed this naming convention from `_ack_listener` and `_nack_listener` from RabbitBasicClass, which have the exact same semantics as `_return_listener` here.